### PR TITLE
MAINTAINERS: add myself to collaborators list

### PR DESCRIPTION
**Pull request by michalsimek**
Add myself to collaborators list for Xilinx/AMD platforms and also cover missing files and folders.

These files are not cover by the fragment:
./drivers/i2c/i2c_xilinx_axi*
./drivers/watchdog/wdt_xilinx_axi.c

And also files in drivers folder which don't contain xlnx like Kconfigs or fpga
./drivers/fpga/fpga_zynqmp*
./drivers/pinctrl/Kconfig.zynqmp

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]
Thanks Michal!

**Review by bjarki-andreasen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@michalsimek In order to be listed as collaborator, you need to nominate yourself as described here: https://docs.zephyrproject.org/latest/project/project_roles.html

**Review by ibirnbaum — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
> > @michalsimek In order to be listed as collaborator, you need to nominate yourself as described here: [docs.zephyrproject.org/latest/project/project_roles.html](https://docs.zephyrproject.org/latest/project/project_roles.html)
> 
> Here it is. #80124

Perfect, then this change should be good to go in once @MaureenHelm has approved it.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
